src/Test/QuickCheck/Classes.hs

```diff
@@ -693,3 +693,49 @@ traversable = const ( "traversable"
 
    fmapP f x = f `fmap` x =-= f `fmapDefault` x
    foldMapP f x = f `foldMap` x =-= f `foldMapDefault` x
+
+foldable :: forall t a b m n o.
+            ( Foldable t, CoArbitrary a, CoArbitrary b, Arbitrary b, Arbitrary (t a), Show b, Show (t a), EqProp b, Monoid m, Arbitrary (t m), Show (t m), EqProp m, Arbitrary (t n), Show (t n), Num n, EqProp n, Arbitrary a, Show a, Eq a, Ord o, Arbitrary (t o), Show (t o), EqProp o, EqProp a) =>
+            t (a, b, m, n, o) -> TestBatch
+foldable = const ( "foldable"
+                 , [ ("foldr and foldMap", property foldrFoldMapP)
+                   , ("foldl and foldMap", property foldlFoldMapP)
+                   , ("fold and foldMap", property foldFoldMapP)
+                   , ("length", property lengthP)
+                   , ("sum", property sumP)
+                   , ("product", property productP)
+                   , ("maximum", property maximumP)
+                   , ("minimum", property minimumP)
+                   , ("foldr1", property foldr1P)
+                   , ("foldl1", property foldl1P)
+                   -- foldr': how to check strictness
+                   -- foldl': how to check strictness
+                   , ("elem", property elemP)
+                   ]
+                 )
+  where
+    foldrFoldMapP :: (a -> b -> b) -> b -> t a -> Property
+    foldrFoldMapP f z t = foldr f z t =-= appEndo (foldMap (Endo . f) t ) z
+    foldlFoldMapP :: (b -> a -> b) -> b -> t a -> Property
+    foldlFoldMapP f z t = foldl f z t =-= appEndo (getDual (foldMap (Dual . Endo . flip f) t)) z
+    foldFoldMapP :: t m -> Property
+    foldFoldMapP t = fold t =-= foldMap id t
```

Comment:
It's been a while since I tinkered with checkers, but I wonder: could we simplify these definitions as follows?
```suggestion
    foldFoldMapP = fold =-= foldMap id
```